ArduinoCode/brake_light_box/brake_light_box.ino

```diff
@@ -0,0 +1,182 @@
+/*
+   Author: Joshua Cheng
+   Date: December 1, 2020
+
+   Pinouts:
+   MOSI - pin 11
+   MISO - pin 12
+   CLK - pin 13
+
+   SD card module:
+   SS - pin 10
+
+   CAN module:
+   SS - pin 9
+
+   Note: This program keeps the CAN module activated by default, but deactivates it when writing to the SD card.
+   This Arduino may miss CAN messages if it recieves a message while writing to the SD card.
+*/
+
+#include <mcp_can.h> // Uses Seeed-studio's CAN_BUS_Shield library.
+#include <mcp_can_dfs.h>
+#include <SPI.h>
+#include <SD.h>
+
+#define SD_SS_PIN 10
+#define CAN_SS_PIN 9
+#define BRAKE_LIGHT_PIN 6 // Does not have to be PWM pin.
+
+File myFile;
+MCP_CAN CAN(CAN_SS_PIN);
+
+unsigned char len = 0; // Length of incoming message
+unsigned char buf[8]; // Memory for incoming message
+int canId; // Sender's ID
+String errorBuffer = "";
+int bufferLength = 0;
+long lastLog = 0; // Time of last successful log to SD
+
+int nextFileName = 0; // Log file names in the format log-0.txt
+
+/**
+   Sets relavent pins to output or input, initializes serial for debugging, initializes CAN, SD card reader,
+   determines proper file name for future logs.
+*/
+void setup() {
+  Serial.begin(9600);
+
+  pinMode(SD_SS_PIN, OUTPUT);
+  pinMode(CAN_SS_PIN, OUTPUT);
+  pinMode(BRAKE_LIGHT_PIN, OUTPUT);
+
+  digitalWrite(SD_SS_PIN, HIGH); // disable SD module
+  digitalWrite(CAN_SS_PIN, LOW); // enable CAN module
+
+  while (CAN_OK != CAN.begin(CAN_500KBPS, MCP_8MHz)) { //specify 8MHz crystal
+    Serial.println("CAN BUS init Failed");
+    delay(250);
+  }
+  Serial.println("CAN BUS Shield Init OK!");
+
+  digitalWrite(CAN_SS_PIN, HIGH); // disable CAN module
+  digitalWrite(SD_SS_PIN, LOW); // enable SD module
```

Comment:
this chunk could be a cool candidate for turning it into a separate `switchCANToSD()` function if you wanted. Totally optional.